### PR TITLE
chore(Cross): [IOAPPX-339] Replace `ButtonExtendedOutline` with `ModuleSummary`

### PR DESCRIPTION
**Pull request by dmnplb**
> [!warning]
> This PR depends on:
> * https://github.com/pagopa/io-app-design-system/pull/305

## Short description
This PR replaces the `ButtonExtendedOutline` with `ModuleSummary`.

## List of changes proposed in this pull request
- Update `io-app-design-system` to `1.40.0`
- Replace `ButtonExtendedOutline` component in the IDPay flow
- Update relative documentation in the **Modules** screen (Design System)

## How to test
1. Launch the app in the local environment
2. Go to the **Design System → Modules** screen

### CODE REVIEW

File: ts/screens/onboarding/OnboardingTosScreen.tsx [hunk 1]
```diff
@@ -142,7 +141,6 @@ const OnboardingTosScreen = () => {
           testID={"currentToSNotAcceptedView"}
         >
           <AlertDS
-            viewRef={viewRef}
```
Comment (CrisTofani) [inline ts/screens/onboarding/OnboardingTosScreen.tsx:145]: what about restoring it as `ref`?

File: ts/features/wallet/component/features/PaymentMethodFeatures.tsx [hunk 2]
```diff
@@ -29,13 +28,8 @@ const PaymentMethodFeatures = ({ paymentMethod }: Props) => {
...
 
   if (isMethodExpired) {
-    const viewRef = React.createRef<View>();
     return (
-      <Alert
-        viewRef={viewRef}
-        variant="error"
-        content={I18n.t("wallet.methodDetails.expired")}
-      />
+      <Alert variant="error" content={I18n.t("wallet.methodDetails.expired")} />
```
Comment (CrisTofani) [inline ts/features/wallet/component/features/PaymentMethodFeatures.tsx:32]: what about restoring the ref?

File: ts/screens/authentication/TestAuthenticationScreen.tsx [hunk 3]
```diff
@@ -67,32 +67,21 @@ const LoadingView = () => (
...
-        content={content}
-      />
-      <VSpacer size={16} />
-    </View>
-  );
-};
+const ErrorView = (title: string, content: string) => (
+  <View style={IOStyles.flex} testID="errorView">
+    <VSpacer size={16} />
+    <Alert variant="error" title={title} content={content} />
```
Comment (CrisTofani) [inline ts/screens/authentication/TestAuthenticationScreen.tsx:73]: ditto

File: ts/screens/authentication/TestAuthenticationScreen.tsx [hunk 4]
```diff
@@ -67,32 +67,21 @@ const LoadingView = () => (
...
+    <VSpacer size={16} />
+  </View>
+);
 
-const SuccessfulView = () => {
-  const viewRef = React.createRef<View>();
-  return (
-    <View style={IOStyles.flex} testID="successView">
-      <VSpacer size={16} />
-      <Alert viewRef={viewRef} variant="success" content={"Success"} />
```
Comment (CrisTofani) [inline ts/screens/authentication/TestAuthenticationScreen.tsx:91]: ditto

**Review by CrisTofani — CHANGES_REQUESTED** [review]
I may have missed some Alert component, generally speaking I guess we can restore the refs as a component ref instead of viewRef prop now removed

**Comment by dmnplb** [comment]
> I may have missed some Alert component, generally speaking I guess we can restore the refs as a component ref instead of viewRef prop now removed

@CrisTofani In most parts of the codebase, the `viewRef` prop was added because it was required by the original component API. Unless it's added voluntarily, I think it's better to get rid of it.

**Review by CrisTofani — APPROVED** [review]

**Pull request merged**